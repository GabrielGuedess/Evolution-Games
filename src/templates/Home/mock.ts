import { AvatarUserProps } from 'components/molecules/AvatarUser/AvatarUser';
import gameCardItemsMock from 'components/organisms/GameList/mock';
import sliderDefaultMock from 'components/organisms/Slider/mock';

export const userInfo: AvatarUserProps = {
  name: 'Gabriel Guedes',
  username: 'gabrielrguedess',
  bio: 'Fullstack Web Developer. Working with Javascript, Typescript, ReactJS, NextJS and NodeJS.',
  userPhoto: 'https://avatars.githubusercontent.com/u/64827875?v=4',
};

export const sliderMock = sliderDefaultMock;

export const gameCardItems = gameCardItemsMock;
