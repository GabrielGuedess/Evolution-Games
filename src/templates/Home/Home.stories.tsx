import { Story, Meta } from '@storybook/react/types-6-0';

import { Home } from './Home';
import { gameCardItems } from './mock';

export default {
  title: 'Templates/Home',
  component: Home,
} as Meta;

export const Default: Story = () => <Home gameList={gameCardItems} />;
