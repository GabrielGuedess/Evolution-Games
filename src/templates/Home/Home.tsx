import { GameList } from 'components/organisms/GameList/GameList';
import { Navbar } from 'components/organisms/Navbar/Navbar';
import { Slider } from 'components/organisms/Slider/Slider';

import * as S from './Home.styles';
import { userInfo, sliderMock, gameCardItems } from './mock';

export const Home = () => (
  <S.Wrapper>
    <Navbar {...userInfo} />
    <Slider slides={sliderMock} />
    <GameList title="Bestsellers" data={gameCardItems} />
    <GameList title="Most Popular Games" data={gameCardItems} />
  </S.Wrapper>
);
