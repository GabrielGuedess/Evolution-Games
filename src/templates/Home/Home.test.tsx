import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { Home } from './Home';

jest.mock('components/organisms/Navbar/Navbar', () => ({
  __esModule: true,
  Navbar: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Navbar">{children}</div>;
  },
}));

jest.mock('components/organisms/Slider/Slider', () => ({
  __esModule: true,
  Slider: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Slider">{children}</div>;
  },
}));

describe('<Home />', () => {
  it('should render the Home correctly', () => {
    // Arrange
    const { container } = renderWithTheme(<Home />);

    // Assert
    expect(screen.getByTestId('Navbar')).toBeInTheDocument();
    expect(screen.getByTestId('Slider')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
