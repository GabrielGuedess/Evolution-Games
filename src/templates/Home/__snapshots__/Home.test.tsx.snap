// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Home /> should render the Home correctly 1'] = `
<main
  class=""
>
  <div
    data-testid="Navbar"
  />
  <div
    data-testid="Slider"
  />
  <div
    data-testid="GameList"
  />
  <div
    data-testid="GameList"
  />
</main>
`;
